hack/test-go.sh

```diff
@@ -181,7 +181,7 @@ else
             -o -path "${kubernetes_path}/test/e2e"                                                    \
             -o -path "${kubernetes_path}/cmd/libs/go2idl/client-gen/testoutput/testgroup/unversioned" \
           \) -prune                                                                                   \
-        \) -name '*_test.go' -print0 | xargs -0n1 dirname -z | sort -uz | xargs -0n1 printf "${OS_GO_PACKAGE}/%s\n")"
+        \) -name '*_test.go' | xargs -n1 dirname | sort -u | xargs -n1 printf "${OS_GO_PACKAGE}/%s\n")"
```

Comment:
The output of the `find` is always relative the the OpenShift root directory, so we would only ever run into this issue if we hosted filenames like that in our repository, and regardless if a directory name in the repository had newlines, or whitespace, *etc*, `dirname` wouldn't have made a difference. Furthermore, once the `dirname` is used without `-z`, the output contains the whitespace and `sort` won't work correctly as it will break on the whitespace, as it's not called with `-z` either. For example, reverting to those flags on the current code in master:
```sh
$ mkdir hello\ world
$ touch hello\ world/some_test.go
$ DRY_RUN=true hack/test-go.sh
The following base flags for `go test` will be used by hack/test-go.sh:
go test  -race -cover -covermode atomic -timeout 60s 
The following packages will be tested by hack/test-go.sh:
github.com/openshift/origin/examples
github.com/openshift/origin/hello
github.com/openshift/origin/world
github.com/openshift/origin/pkg/api
...
```